client/k8srestconfig/k8s_rest_config.go

```diff
@@ -124,7 +127,15 @@ func New(config Config) (*rest.Config, error) {
 	var err error
 
 	var restConfig *rest.Config
-	if config.InCluster {
+	if config.KubeConfig != "" {
+		config.Logger.Log("level", "debug", "message", "using kubeconfig")
```

Comment:
Does something like this make sense? We should then align the logs below accordingly. 

```
creating rest config using in-cluster flag
creating rest config using out-cluster flag
```

```suggestion
		config.Logger.Log("level", "debug", "message", "creating rest config using kubeconfig flag")
```

Reply:
We should maybe also have the closing logs like anywhere else when we have the opening logs.

Reply:
Good point. I'll align the logging and add the closing logs.